Naveen-gali/Yelp-App
# Firebase integration

### What does this PR do?

- Adds Firebase to the App
- Adds Crashlytics and Firebase Analytics to App.
- Adds Google Authentication to App.
- Adds Performance Monitoring to App.
- Adds Support for Cloud Messaging to App.
- Adds Permissions to App.
- Adds Profile Updation Feature.
- Adds Shimmer Effect to HomeScreen Featured Categories.


### Related Issues

- closes #51 
- closes #52 
- closes #53 
- closes #54 

### How to test

- Clone the Repository.
- Switch to feat/FirebaseIntegration branch.
- Do yarn to install the dependencies.
- Do yarn run **build-stage-ios** for ios or **yarn run build-stage-android** for android.
- While Opening app, It will ask for Google Signin.
- Navigate to Profile or Me tab and click on the profile image at top of the screen.
- Now in bottom sheet, you can choose one of the 2 methods Camera or Storage.
- It will ask for respective permissions. If allowed will ask to choose for photo or capture a photo.
- On successfully selecting will upload the photo to storage and you can see your new profile photo on the top.


### Recordings


#### Android Recording 

https://github.com/Naveen-gali/Yelp-App/assets/122969303/b2a6dbc8-4fe1-43b2-96b8-8c4e6d9ac417

#### Ios Recording

https://github.com/Naveen-gali/Yelp-App/assets/122969303/0cb6ca34-b66b-4215-a1e8-4da46cf62b09

### Checklist
- [x] I have addressed all the requirements.
- [x] I have self reviewed the PR.




### Packages Added (if any)

| Package | Reason |
| ------- | ------ |
| @gorhom/bottom-sheet        |  To Add Bottom Sheet       |
|@react-native-community/push-notification-ios | Push Notification Service Package for IOS |
|@react-native-firebase/analytics| Analytics Support for Firebase |
|@react-native-firebase/app | Core Firebase Package |
|@react-native-firebase/auth | Authentication Support Package |
|@react-native-firebase/crashlytics |  Crashlytics Support Package |
|@react-native-firebase/messaging | Cloud Messaging Support Package |
|@react-native-firebase/perf | Performance Monitoring Support Package |
|@react-native-firebase/remote-config | Remote Config Package |
|@react-native-firebase/storage | Storage Support |
|@react-native-google-signin/google-signin | Google Signin Support Package for Firebase |
| react-native-skeleton-placeholder | Skeleton Placeholder |
| react-native-image-crop-picker | Image Picker |
| react-native-linear-gradient | Linear Gradient |
| react-native-permissions | App Permissions |
| react-native-push-notification | Push Notifications |
